src/main/kotlin/io/meshcloud/dockerosb/metrics/gauge/PagedGaugeController.kt

```diff
@@ -0,0 +1,53 @@
+package io.meshcloud.dockerosb.metrics.gauge
+
+import io.meshcloud.dockerosb.metrics.MetricsProvider
+import io.meshcloud.dockerosb.metrics.MetricsResponse
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.time.Instant
+
+
+@RestController
+class PagedGaugeController(
+    private val metricsProviders: List<MetricsProvider<GaugeMetricModel>>
+) {
+
+  @GetMapping("/metrics/gauges/{serviceDefinitionId}")
+  fun getGaugeMetricValues(
+      @PathVariable("serviceDefinitionId") serviceDefinitionId: String,
+      @RequestParam from: Instant,
+      @RequestParam to: Instant
+  ): ResponseEntity<MetricsResponse<GaugeMetricModel>> {
+    return getResponse(serviceDefinitionId, from, to, 0)
+  }
+
+  @GetMapping("/metrics/gauges/{serviceDefinitionId}/{index}")
```

Comment:
d: so the idea is to fetch one by one, does it make sense like this or should we provide proper pagination with customizable page size and page number?

Reply:
i will add this feature later. it is totally possible to extend the metrics feature with proper pagination. we need also adapt new RequestParams like size and skip(nextMarker), so the next feature implementation will have backward compatibility and people can still use instance index if needed. for now the time-filters should work for filtering the huge datas.